gammapy/utils/docs.py

```diff
@@ -96,3 +100,81 @@ def gammapy_sphinx_ext_activate():
     # Register our directives and roles with Sphinx
     register_directive('gp-extra-image', ExtraImage)
     roles.register_local_role('gp-extra-notebook', notebook_role)
+
+def modif_nb_links(folder, url_docs):
+    """
+    Modifies links in raw and sphinx formatted notebooks and so they
+    point to and from the same version of the documentation. Adds a box to the
+    sphinx formatted notebooks with info and link to the ipynb file.
+    """
+
+    DOWNLOAD_CELL = """
+<div class='admonition note'>
+**This is a fixed-text formatted version of a Jupyter notebook.**
+
+You can download for each version of *gammapy* a
+[HTMLZip pack](http://readthedocs.org/projects/gammapy/downloads/) containing
+the whole HTML documentation and full collection of notebooks, so you can execute
+them in your local `_static/notebooks/` folder. You can also contribute with your
+own notebooks in this [GitHub repository](https://github.com/gammapy/gammapy-extra/tree/master/notebooks).
+
+**Source files:**
+[{nb_filename}](../_static/notebooks/{nb_filename}) |
+[{py_filename}](../_static/notebooks/{txt_filename})
+</div>"""
+
+    for filename in os.listdir(folder):
+        filepath = os.path.join(folder, filename)
+        if os.path.isfile(filepath) and filepath[-6:] == '.ipynb':
+            if folder=='notebooks':
+                py_filename = filename.replace('ipynb', 'py')
+                txt_filename = filename.replace('ipynb', 'txt')
+                ctx = dict(nb_filename=filename, py_filename=py_filename, txt_filename=txt_filename)
+                strcell = DOWNLOAD_CELL.format(**ctx)
+                nb = nbformat.read(filepath, as_version=nbformat.NO_CONVERT)
+                nb.cells.insert(0, new_markdown_cell(strcell))
+                nbformat.write(nb, filepath)
+            with open(filepath) as f:
+                txt = f.read()
+            if folder=='notebooks':
+                txt = re.sub(url_docs+'(.*?)html(\)|#)',r'..\1rst\2', txt, flags=re.M|re.I)
+            if folder=='_static/notebooks':
+                txt = re.sub(url_docs+'(.*?)html(\)|#)',r'..\/..\1html\2', txt, flags=re.M|re.I)
+            with open(filepath, "w") as f:
+                f.write(txt)
+
+def replace_extension(folder):
+    """
+    Replaces extension of .py files so they can be served for download
+    """
+
+    for filename in os.listdir(folder):
+        txtfilename = filename.replace('.py', '.txt')
+        os.rename(os.path.join(folder, filename), os.path.join(folder, txtfilename))
+
+def gammapy_sphinx_notebooks(setup_cfg):
+    """
+    Manages the processes for the building of sphinx formatted notebooks
+    """
+
+    url_docs = setup_cfg.get('url_docs')
```

Comment:
Suggest to change to `url_docs = setup_cfg['url_docs']`.

The difference is that `dict.get` will return `None` by default. And if you put `None` here in `url_docs`, I think your code will error much later with a `TypeError`.

Better to fail early and clearly. And in this case, the `url_docs` key should always be present, so no need to implement special code for the case where it's not present.